<a href="https://colab.research.google.com/github/Sudo343/Big-Data-Analytics/blob/main/B_D_Project.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

Submitted by:  
  Suddhodan Katolkar, 212CD026  
  Dinesh Ladi, 212CD009  
  P V Tejeswar Reddy, 212CD013


# Big Data Project  
Scopus refine results values  

  Your query : KEY ( "internet of things"  ,  "wireless sensor networks" )  AND  ( LIMIT-TO ( SUBJAREA ,  "ENGI" )  OR  LIMIT-TO ( SUBJAREA ,  "MEDI" ) )  AND  ( LIMIT-TO ( PUBSTAGE ,  "final" ) )  AND  ( LIMIT-TO ( DOCTYPE ,  "ar" ) )  AND  ( LIMIT-TO ( SUBJAREA ,  "CHEM" ) )  AND  ( LIMIT-TO ( SRCTYPE ,  "j" ) )  							

  Keyword:       "internet of things"  ,  "wireless sensor networks"  
  Document Type: Article  
  Subject Area:  Engineering, Medicine, Chemistry  
  Source Type:   Journal  

  Number of results : 314

In [ ]:
import pandas as pd

In [ ]:
df = pd.read_csv('/content/scopus_2.csv')

In [ ]:
df.head()

Authors  \
0                     Li M., Fang C.-C., Ferng H.-W.   
1            Jecan E., Pop C., Ratiu O., Puschita E.   
2                    Rzymowski M., Nyka K., Kulas L.   
3  Majid M., Habib S., Javed A.R., Rizwan M., Sri...   
4             Tsai R.-G., Lv X., Shen L., Tsai P.-H.   

                                        Author(s) ID  \
0               57540025700;57219573519;35614169400;   
1   57202972685;57202972136;57193877852;14632312300;   
2                 55328900100;6506332547;6603319516;   
3  57226481291;56432246900;57215024733;5722049013...   
4   55616405200;57479957300;57480648300;16508092400;   

                                               Title  Year Source title  \
0  On-Demand Energy Transfer and Energy-Aware Pol...  2022      Sensors   
1  Predictive Energy-Aware Routing Solution for I...  2022      Sensors   
2  Direction of Arrival Estimation Based on Recei...  2022      Sensors   
3  Applications of Wireless Sensor Networks and I...  2022      Sensors   
4  A High-Robust Sensor Activity Control Algorith...  2022      Sensors   

   Cited by                DOI  \
0       NaN  10.3390/s22072476   
1       NaN  10.3390/s22062107   
2       NaN  10.3390/s22052034   
3       NaN  10.3390/s22062087   
4       NaN  10.3390/s22052020   

                                                Link  \
0  https://www.scopus.com/inward/record.uri?eid=2...   
1  https://www.scopus.com/inward/record.uri?eid=2...   
2  https://www.scopus.com/inward/record.uri?eid=2...   
3  https://www.scopus.com/inward/record.uri?eid=2...   
4  https://www.scopus.com/inward/record.uri?eid=2...   

                                        Affiliations  \
0  Department of Electrical Engineering, School o...   
1  Communications Department, Technical Universit...   
2  Department of Microwave and Antenna Engineerin...   
3  School of System and Technology, University of...   
4  New Engineering Industry College, Putian Unive...   

                           Authors with affiliations  \
0  Li, M., Department of Electrical Engineering, ...   
1  Jecan, E., Communications Department, Technica...   
2  Rzymowski, M., Department of Microwave and Ant...   
3  Majid, M., School of System and Technology, Un...   
4  Tsai, R.-G., New Engineering Industry College,...   

                               Funding Details Document Type  \
0                                          NaN       Article   
1                                          NaN       Article   
2                                       876038       Article   
3                                          NaN       Article   
4  JAT190580, JAT210410; 2020J01924, JAT190573       Article   

  Publication Stage                   Open Access  Source                 EID  
0             Final  All Open Access, Gold, Green  Scopus  2-s2.0-85126799378  
1             Final  All Open Access, Gold, Green  Scopus  2-s2.0-85126064749  
2             Final  All Open Access, Gold, Green  Scopus  2-s2.0-85125962511  
3             Final  All Open Access, Gold, Green  Scopus  2-s2.0-85125951647  
4             Final  All Open Access, Gold, Green  Scopus  2-s2.0-85125912246

In [ ]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 314 entries, 0 to 313
Data columns (total 16 columns):
 #   Column                     Non-Null Count  Dtype  
---  ------                     --------------  -----  
 0   Authors                    314 non-null    object 
 1   Author(s) ID               314 non-null    object 
 2   Title                      314 non-null    object 
 3   Year                       314 non-null    int64  
 4   Source title               314 non-null    object 
 5   Cited by                   278 non-null    float64
 6   DOI                        314 non-null    object 
 7   Link                       314 non-null    object 
 8   Affiliations               314 non-null    object 
 9   Authors with affiliations  314 non-null    object 
 10  Funding Details            176 non-null    object 
 11  Document Type              314 non-null    object 
 12  Publication Stage          314 non-null    object 
 13  Open Access                308 non-null    object 

In [ ]:
df.isnull().sum()

Authors                        0
Author(s) ID                   0
Title                          0
Year                           0
Source title                   0
Cited by                      36
DOI                            0
Link                           0
Affiliations                   0
Authors with affiliations      0
Funding Details              138
Document Type                  0
Publication Stage              0
Open Access                    6
Source                         0
EID                            0
dtype: int64

In [ ]:
df['Author(s) ID'].head()

0                 57540025700;57219573519;35614169400;
1     57202972685;57202972136;57193877852;14632312300;
2                   55328900100;6506332547;6603319516;
3    57226481291;56432246900;57215024733;5722049013...
4     55616405200;57479957300;57480648300;16508092400;
Name: Author(s) ID, dtype: object

In [ ]:
for i in df['Authors'].head() :
  print(i.split(';'))

['Li M., Fang C.-C., Ferng H.-W.']
['Jecan E., Pop C., Ratiu O., Puschita E.']
['Rzymowski M., Nyka K., Kulas L.']
['Majid M., Habib S., Javed A.R., Rizwan M., Srivastava G., Gadekallu T.R., Lin J.C.-W.']
['Tsai R.-G., Lv X., Shen L., Tsai P.-H.']


In [ ]:
ids = []
Names = set({})
entries = []
for i in df['Author(s) ID'] :
  count =0
  for n in i.split(';') :
    if n is not '' :
      ids.append(n)
      count +=1
  entries.append(count)
# ids.remove('')
# print(ids)

In [ ]:
 len(entries)

314

In [ ]:
len(ids)

1393

In [ ]:
nameList = []
for names in df['Authors'] :
  N = names.split(', ')
  for i in range(0,len(N),1):
    # if i+1 == len(N) :
    nameList.append(N[i])
    # else :
      # nameList.append(N[i]+' '+N[i+1])
#     ids.add(n)
# ids.discard('')
for name in nameList :
  name = name.split(' ',1)
# print(nameList)

In [ ]:
len(nameList)

1393

In [ ]:
df['Affiliations'].head()

0    Department of Electrical Engineering, School o...
1    Communications Department, Technical Universit...
2    Department of Microwave and Antenna Engineerin...
3    School of System and Technology, University of...
4    New Engineering Industry College, Putian Unive...
Name: Affiliations, dtype: object

Now to build a table for  authors

see that rows have multiple values, let's make another table for authors. 
/// *** add country column, for this just check author's address or affiliation, 

In [ ]:

Dict = {}
Dict['Author(s) ID'] = ids
Dict['Author'] = nameList
# Dict['University'] = Uninames
# Dict['Country'] = country
# print(Dict)

In [ ]:
print(len(ids),len(nameList))

1393 1393


In [ ]:
Author_table = pd.DataFrame.from_dict( Dict )

In [ ]:
Author_table

Author(s) ID               Author
0     57540025700                Li M.
1     57219573519           Fang C.-C.
2     35614169400          Ferng H.-W.
3     57202972685             Jecan E.
4     57202972136               Pop C.
...           ...                  ...
1388  54406535200           Bleda A.L.
1389  35113243800            Jara A.J.
1390   6603109177           Maestre R.
1391  54585968100             Santa G.
1392   7004419066  Gómez Skarmeta A.F.

[1393 rows x 2 columns]

In [ ]:
Author_table.drop_duplicates()

Author(s) ID               Author
0     57540025700                Li M.
1     57219573519           Fang C.-C.
2     35614169400          Ferng H.-W.
3     57202972685             Jecan E.
4     57202972136               Pop C.
...           ...                  ...
1388  54406535200           Bleda A.L.
1389  35113243800            Jara A.J.
1390   6603109177           Maestre R.
1391  54585968100             Santa G.
1392   7004419066  Gómez Skarmeta A.F.

[1261 rows x 2 columns]

Now that we have author table, we can drop the names from df and just use the Author(s) ID column to resolve the multiple values of df table.

In [ ]:
 df_1 = df[ df.columns[1:]]
df_1.shape

(314, 15)

In [ ]:
Dict = {} 
n, m =  df_1.shape[0], df_1.shape[1]
ids = df_1['Author(s) ID']
Lists = [ [] for i in range(m)]
for ind in range(n) :
  Id = ids[ind].split(';')
  for k in Id :
    if k is not '' :
      Lists[0].append(k)
      for j in range(1,m) :
        Lists[j].append(df_1.iloc[ind,j])

In [ ]:
ids = []
for i in df['Author(s) ID'] :
  for n in i.split(';') :
    if n is not '' :
      ids.append(n)
# ids.remove('')
# print(ids)

In [ ]:
ids = df_1['Author(s) ID']
# ids

In [ ]:
for j in range(df_1.columns.size) :
  Dict[df_1.columns[j]] = Lists[j]
# Dict

In [ ]:
df_final = pd.DataFrame.from_dict(Dict)
df_final.head(25)

Author(s) ID                                              Title  Year  \
0   57540025700  On-Demand Energy Transfer and Energy-Aware Pol...  2022   
1   57219573519  On-Demand Energy Transfer and Energy-Aware Pol...  2022   
2   35614169400  On-Demand Energy Transfer and Energy-Aware Pol...  2022   
3   57202972685  Predictive Energy-Aware Routing Solution for I...  2022   
4   57202972136  Predictive Energy-Aware Routing Solution for I...  2022   
5   57193877852  Predictive Energy-Aware Routing Solution for I...  2022   
6   14632312300  Predictive Energy-Aware Routing Solution for I...  2022   
7   55328900100  Direction of Arrival Estimation Based on Recei...  2022   
8    6506332547  Direction of Arrival Estimation Based on Recei...  2022   
9    6603319516  Direction of Arrival Estimation Based on Recei...  2022   
10  57226481291  Applications of Wireless Sensor Networks and I...  2022   
11  56432246900  Applications of Wireless Sensor Networks and I...  2022   
12  57215024733  Applications of Wireless Sensor Networks and I...  2022   
13  57220490131  Applications of Wireless Sensor Networks and I...  2022   
14  57202588447  Applications of Wireless Sensor Networks and I...  2022   
15  57312879900  Applications of Wireless Sensor Networks and I...  2022   
16  56449520400  Applications of Wireless Sensor Networks and I...  2022   
17  55616405200  A High-Robust Sensor Activity Control Algorith...  2022   
18  57479957300  A High-Robust Sensor Activity Control Algorith...  2022   
19  57480648300  A High-Robust Sensor Activity Control Algorith...  2022   
20  16508092400  A High-Robust Sensor Activity Control Algorith...  2022   
21  57465957900  High Density Resistive Array Readout System fo...  2022   
22  57465081100  High Density Resistive Array Readout System fo...  2022   
23  57194413882  High Density Resistive Array Readout System fo...  2022   
24   8278651200  High Density Resistive Array Readout System fo...  2022   

   Source title  Cited by                DOI  \
0       Sensors       NaN  10.3390/s22072476   
1       Sensors       NaN  10.3390/s22072476   
2       Sensors       NaN  10.3390/s22072476   
3       Sensors       NaN  10.3390/s22062107   
4       Sensors       NaN  10.3390/s22062107   
5       Sensors       NaN  10.3390/s22062107   
6       Sensors       NaN  10.3390/s22062107   
7       Sensors       NaN  10.3390/s22052034   
8       Sensors       NaN  10.3390/s22052034   
9       Sensors       NaN  10.3390/s22052034   
10      Sensors       NaN  10.3390/s22062087   
11      Sensors       NaN  10.3390/s22062087   
12      Sensors       NaN  10.3390/s22062087   
13      Sensors       NaN  10.3390/s22062087   
14      Sensors       NaN  10.3390/s22062087   
15      Sensors       NaN  10.3390/s22062087   
16      Sensors       NaN  10.3390/s22062087   
17      Sensors       NaN  10.3390/s22052020   
18      Sensors       NaN  10.3390/s22052020   
19      Sensors       NaN  10.3390/s22052020   
20      Sensors       NaN  10.3390/s22052020   
21      Sensors       1.0  10.3390/s22051878   
22      Sensors       1.0  10.3390/s22051878   
23      Sensors       1.0  10.3390/s22051878   
24      Sensors       1.0  10.3390/s22051878   

                                                 Link  \
0   https://www.scopus.com/inward/record.uri?eid=2...   
1   https://www.scopus.com/inward/record.uri?eid=2...   
2   https://www.scopus.com/inward/record.uri?eid=2...   
3   https://www.scopus.com/inward/record.uri?eid=2...   
4   https://www.scopus.com/inward/record.uri?eid=2...   
5   https://www.scopus.com/inward/record.uri?eid=2...   
6   https://www.scopus.com/inward/record.uri?eid=2...   
7   https://www.scopus.com/inward/record.uri?eid=2...   
8   https://www.scopus.com/inward/record.uri?eid=2...   
9   https://www.scopus.com/inward/record.uri?eid=2...   
10  https://www.scopus.com/inward/record.uri?eid=2...   
11  https://www.scopus.com/inward/record.uri?eid=2...   
12  https://www.scopus.com/inward/re

In [ ]:
df_final.shape

(1393, 15)

Now resolve the affiliations in table

In [ ]:
aff = []
# aff = list(df_final['Affiliations'])  # 'Authors with affiliations'
for row in df_final['Affiliations'] :
  # if row not in aff :
    aff.append(row)
len(aff)

1393

In [ ]:
# UniList = []
# for i in range(len(aff)) :
#   A = aff[i].split('; ')
#   skip = len(A)
#   if skip > 1 :
#     ind=0
#     for ind in range(skip) :
#       UniList.append( A[ind])
#       i +=1
#   else :
#     UniList.append( A[0])
# len(UniList)
# # UniList
    

In [ ]:
for i in range(len(aff)) :
  corr = aff[i].split('; ')
  size= len(corr)
  id=0
  while size > 1 and id<size : 
    # so there are corrections to be made
    aff[i+id] = corr[id]
    id = id+1
  i = i+id

len(aff)

1393

In [ ]:
country = []
for row in aff :
  l = row.split(', ')
  country.append(l[-1])
len(country)

1393

In [ ]:
for j in range(m) :
  Dict[df_1.columns[j]] = Lists[j]
Dict['University'] = aff
Dict['Country'] = country
# Dict

In [ ]:
df_final_1 = pd.DataFrame.from_dict(Dict)
df_final_1.head()

Author(s) ID                                              Title  Year  \
0  57540025700  On-Demand Energy Transfer and Energy-Aware Pol...  2022   
1  57219573519  On-Demand Energy Transfer and Energy-Aware Pol...  2022   
2  35614169400  On-Demand Energy Transfer and Energy-Aware Pol...  2022   
3  57202972685  Predictive Energy-Aware Routing Solution for I...  2022   
4  57202972136  Predictive Energy-Aware Routing Solution for I...  2022   

  Source title  Cited by                DOI  \
0      Sensors       NaN  10.3390/s22072476   
1      Sensors       NaN  10.3390/s22072476   
2      Sensors       NaN  10.3390/s22072476   
3      Sensors       NaN  10.3390/s22062107   
4      Sensors       NaN  10.3390/s22062107   

                                                Link  \
0  https://www.scopus.com/inward/record.uri?eid=2...   
1  https://www.scopus.com/inward/record.uri?eid=2...   
2  https://www.scopus.com/inward/record.uri?eid=2...   
3  https://www.scopus.com/inward/record.uri?eid=2...   
4  https://www.scopus.com/inward/record.uri?eid=2...   

                                        Affiliations  \
0  Department of Electrical Engineering, School o...   
1  Department of Electrical Engineering, School o...   
2  Department of Electrical Engineering, School o...   
3  Communications Department, Technical Universit...   
4  Communications Department, Technical Universit...   

                           Authors with affiliations Funding Details  \
0  Li, M., Department of Electrical Engineering, ...             NaN   
1  Li, M., Department of Electrical Engineering, ...             NaN   
2  Li, M., Department of Electrical Engineering, ...             NaN   
3  Jecan, E., Communications Department, Technica...             NaN   
4  Jecan, E., Communications Department, Technica...             NaN   

  Document Type Publication Stage                   Open Access  Source  \
0       Article             Final  All Open Access, Gold, Green  Scopus   
1       Article             Final  All Open Access, Gold, Green  Scopus   
2       Article             Final  All Open Access, Gold, Green  Scopus   
3       Article             Final  All Open Access, Gold, Green  Scopus   
4       Article             Final  All Open Access, Gold, Green  Scopus   

                  EID                                         University  \
0  2-s2.0-85126799378  Department of Electrical Engineering, School o...   
1  2-s2.0-85126799378  Department of Computer Science and Information...   
2  2-s2.0-85126799378  Department of Electrical Engineering, School o...   
3  2-s2.0-85126064749  Department of Computer Science and Information...   
4  2-s2.0-85126064749  Communications Department, Technical Universit...   

   Country  
0   Taiwan  
1   Taiwan  
2   Taiwan  
3   Taiwan  
4  Romania

In [ ]:
df_final_1.drop( ['Affiliations','Authors with affiliations'], axis=1, inplace=True)

In [ ]:
df_final_1.loc[:5,['Author(s) ID','University','Country']]
# drop affiliations 

Author(s) ID                                         University  Country
0  57540025700  Department of Electrical Engineering, School o...   Taiwan
1  57219573519  Department of Computer Science and Information...   Taiwan
2  35614169400  Department of Electrical Engineering, School o...   Taiwan
3  57202972685  Department of Computer Science and Information...   Taiwan
4  57202972136  Communications Department, Technical Universit...  Romania
5  57193877852  Control Data Systems S.R.L., Liberty Technolog...  Romania

In [ ]:
# save this table just in case...
df_final_1.to_csv('SCOPUS_final.csv')

so, finally we have proper table to work with.

In [ ]:
df_final_1.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1393 entries, 0 to 1392
Data columns (total 15 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   Author(s) ID       1393 non-null   object 
 1   Title              1393 non-null   object 
 2   Year               1393 non-null   int64  
 3   Source title       1393 non-null   object 
 4   Cited by           1245 non-null   float64
 5   DOI                1393 non-null   object 
 6   Link               1393 non-null   object 
 7   Funding Details    835 non-null    object 
 8   Document Type      1393 non-null   object 
 9   Publication Stage  1393 non-null   object 
 10  Open Access        1367 non-null   object 
 11  Source             1393 non-null   object 
 12  EID                1393 non-null   object 
 13  University         1393 non-null   object 
 14  Country            1393 non-null   object 
dtypes: float64(1), int64(1), object(13)
memory usage: 163.4+ KB


Q1.  

### a) Highest cited author and his h-index (from the world)

In [ ]:
all_cited = df_final_1.groupby(by=['Author(s) ID']).sum(['Cited by'])
all_cited.sort_values('Cited by',ascending=False, inplace= True)
all_cited['Cited by']

Author(s) ID
56039568800    298.0
24467756100    298.0
57189299510    278.0
7006010714     183.0
7003391364     183.0
               ...  
6604043456       0.0
6701608009       0.0
6701640053       0.0
36195020800      0.0
57210010827      0.0
Name: Cited by, Length: 1245, dtype: float64

In [ ]:
ans1,ans2 = all_cited.index[0],all_cited.index[1]
ans1,ans2

('56039568800', '24467756100')

we have the index, now get the name

In [ ]:
Author_table[ Author_table['Author(s) ID'] == ans1 ]
# .loc[:,'Author']

Author(s) ID          Author
305   56039568800  Fraga-Lamas P.
642   56039568800  Fraga-Lamas P.
915   56039568800  Fraga-Lamas P.
940   56039568800  Fraga-Lamas P.
1126  56039568800  Fraga-Lamas P.

In [ ]:
Author_table[ Author_table['Author(s) ID'] == ans2 ]
# .loc[:,'Author']

Author(s) ID                  Author
310   24467756100  Fernández-Caramés T.M.
650   24467756100  Fernández-Caramés T.M.
914   24467756100  Fernández-Caramés T.M.
939   24467756100  Fernández-Caramés T.M.
1127  24467756100  Fernández-Caramés T.M.

So, the most cited author for this dataset are Fraga-Lamas P., Fernández-Caramés T.M. . 

In [ ]:
# calculate h-index 
d2 = df_final_1[df_final_1['Author(s) ID']==ans1].loc[:,['Title','Cited by']].sort_values( by='Cited by', ascending =False)
d2 = pd.DataFrame(d2)
d2.reset_index()

index                                              Title  Cited by
0   1126  Towards the internet of smart trains: A review...     117.0
1    915  Design, implementation and practical evaluatio...      96.0
2    940  A cost-effective IoT system for monitoring ind...      37.0
3    642  Design and experimental validation of a lorawa...      28.0
4    305  Design, implementation, and empirical validati...      20.0

In [ ]:
# calculate h-index 
d2 = df_final_1[df_final_1['Author(s) ID']==ans2].loc[:,['Title','Cited by']].sort_values( by='Cited by', ascending =False)
d2 = pd.DataFrame(d2)
d2.reset_index()

index                                              Title  Cited by
0   1127  Towards the internet of smart trains: A review...     117.0
1    914  Design, implementation and practical evaluatio...      96.0
2    939  A cost-effective IoT system for monitoring ind...      37.0
3    650  Design and experimental validation of a lorawa...      28.0
4    310  Design, implementation, and empirical validati...      20.0

5 publication for both the authors,  
so from above table, h-index is 5.  

In this dataset we see that both these authors worked together on these five articles.  

on checking with world, we get h-index=28  (Fraga-Lamas P.)  
https://www.scopus.com/authid/detail.uri?origin=resultslist&authorId=56039568800&zone=  
 
we get h-index=30 (Fernández-Caramés T.M.)  
https://www.scopus.com/authid/detail.uri?origin=resultslist&authorId=24467756100&zone=

### b)  Highest publication author

In [ ]:
# get author with most publications
ans2 = df_final_1.groupby(by=['Author(s) ID']).count()['Publication Stage'].sort_values( ascending=False).index[0]
ans2

'24467846500'

In [ ]:
Author_table[ Author_table['Author(s) ID'] == ans2 ]

Author(s) ID      Author
309  24467846500  Falcone F.
592  24467846500  Falcone F.
632  24467846500  Falcone F.
649  24467846500  Falcone F.
923  24467846500  Falcone F.

here, we have author : 24467846500, 	Falcone F. 

### c) Highest cited authors avg. citations, and the country name.

In [ ]:
# first see for avg. citations
# ans_c = df_final_1.groupby(['Author(s) ID']).mean('Cited by').sort_values(by=['Cited by'], ascending=False).index[0]
# ans_c
ans_c = df_final_1.groupby(['Author(s) ID']).mean('Cited by').sort_values(by=['Cited by'], ascending=False)
ans_c

Year  Cited by
Author(s) ID                  
7003391364    2016.0     183.0
7006010714    2016.0     183.0
24824284200   2016.0     183.0
53363838700   2016.0     183.0
57191954527   2016.0     183.0
...              ...       ...
7004047183    2022.0       NaN
7005884899    2021.0       NaN
7404945944    2019.0       NaN
8338455600    2013.0       NaN
8661455300    2013.0       NaN

[1245 rows x 2 columns]

In [ ]:
# ans_c[['56039568800','24467756100']:]
a1 = ans_c['56039568800':]

In [ ]:
a1.iloc[0,:]

Year        2018.4
Cited by      59.6
Name: 56039568800, dtype: float64

In [ ]:
a2 = ans_c['24467756100':]

In [ ]:
a2.iloc[0,:]

Year        2018.4
Cited by      59.6
Name: 24467756100, dtype: float64

In [ ]:
# Author_table[Author_table['Author(s) ID']==ans_c]
Author_table[Author_table['Author(s) ID']=='56039568800']

Author(s) ID          Author
305   56039568800  Fraga-Lamas P.
642   56039568800  Fraga-Lamas P.
915   56039568800  Fraga-Lamas P.
940   56039568800  Fraga-Lamas P.
1126  56039568800  Fraga-Lamas P.

In [ ]:
df_final_1[df_final_1['Author(s) ID']=='56039568800']['Country']

305     Spain
642     China
915     Spain
940     Spain
1126    Spain
Name: Country, dtype: object

In [ ]:
# Author_table[Author_table['Author(s) ID']==ans_c]
Author_table[Author_table['Author(s) ID']=='24467756100']

Author(s) ID                  Author
310   24467756100  Fernández-Caramés T.M.
650   24467756100  Fernández-Caramés T.M.
914   24467756100  Fernández-Caramés T.M.
939   24467756100  Fernández-Caramés T.M.
1127  24467756100  Fernández-Caramés T.M.

In [ ]:
df_final_1[df_final_1['Author(s) ID']=='56039568800']['Country']

305     Spain
642     China
915     Spain
940     Spain
1126    Spain
Name: Country, dtype: object

They are (24467756100) Fernández-Caramés T.M. & (56039568800) Fraga-Lamas P. are the highest cited authors in this dataset, their avg. citations is 59.6, they worked together on same papers while they were in Spain & China.

### d) Total number of publications of the highest cited author

In [ ]:
ans3 = df_final_1.groupby(by=['Author(s) ID']).nunique(['Title'])

ans3[ans3.index == ans1]['Title']

Author(s) ID
56039568800    5
Name: Title, dtype: int64

so, 5 publications, by highest cited author.  

### e) Total publication in year

In [ ]:
df.groupby(['Year']).count()

Authors  Author(s) ID  Title  Source title  Cited by  DOI  Link  \
Year                                                                    
2012        2             2      2             2         2    2     2   
2013        3             3      3             3         2    3     3   
2014       12            12     12            12        11   12    12   
2015       13            13     13            13        13   13    13   
2016       21            21     21            21        21   21    21   
2017       25            25     25            25        25   25    25   
2018       51            51     51            51        51   51    51   
2019       59            59     59            59        56   59    59   
2020       63            63     63            63        60   63    63   
2021       51            51     51            51        35   51    51   
2022       14            14     14            14         2   14    14   

      Affiliations  Authors with affiliations  Funding Details  Document Type  \
Year                                                                            
2012             2                          2                1              2   
2013             3                          3                0              3   
2014            12                         12                4             12   
2015            13                         13                1             13   
2016            21                         21                8             21   
2017            25                         25               14             25   
2018            51                         51               35             51   
2019            59                         59               36             59   
2020            63                         63               40             63   
2021            51                         51               29             51   
2022            14                         14                8             14   

      Publication Stage  Open Access  Source  EID  
Year                                               
2012                  2            2       2    2  
2013                  3            3       3    3  
2014                 12           12      12   12  
2015                 13           12      13   13  
2016                 21           19      21   21  
2017                 25           25      25   25  
2018                 51           50      51   51  
2019                 59           59      59   59  
2020                 63           62      63   63  
2021                 51           50      51   51  
2022                 14           14      14   14

In [ ]:
df[ df['Publication Stage']=='Final' ].groupby(['Year']).count().loc[:,'Publication Stage']

Year
2012     2
2013     3
2014    12
2015    13
2016    21
2017    25
2018    51
2019    59
2020    63
2021    51
2022    14
Name: Publication Stage, dtype: int64

Here in above table, we have number of publications that were finalized every year.  

### f) Total citation per year

In [ ]:
df.groupby(['Year']).sum(['Cited by'])

Cited by
Year          
2012      30.0
2013     145.0
2014     235.0
2015     459.0
2016     507.0
2017     928.0
2018    1302.0
2019     974.0
2020     687.0
2021     150.0
2022       2.0

Here in above table, we have number of citations every year.  

### g) Author(country) having highest co-authorship with indian authors.


In [ ]:
# first group the papers by title, then choose ones with indians authors, then find
ind_auth_list = list(df_final_1[ df_final_1['Country']=='India']['Author(s) ID'])
# ind_auth_list

In [ ]:
# cnt=[]
# for i in range(df.shape[0]):
#     a = len(df['Author(s) ID'][i].split(','))
#     b = len(set(df['Author(s) ID'][i].split(','))-set(ind_auth_list))
#     if(b!=a):
#         cnt.append(1)
#     else :
#         cnt.append(0)

In [ ]:
co_authors = []
for i in range(df.shape[0]):
    k = df['Author(s) ID'][i].split(',')
    for j in k:
        if j in ind_auth_list:
            continue
        else:
            co_authors.append(j)

In [ ]:
import collections
freq = collections.Counter(co_authors)
#print(dict(freq))
freq = dict(freq)
# freq

In [ ]:
res = 0
for i in freq.items():
    if i[1]>res:
        res = i[1]
res

5

In [ ]:
for i in freq.items():
    if i[1]==res:
        print(i[0])

57195998372;56812550600;


In [ ]:
Author_table[Author_table['Author(s) ID']=='57195998372']

Author(s) ID      Author
35   57195998372  Dinh N.-T.
45   57195998372     Dinh N.
90   57195998372  Dinh N.-T.
301  57195998372  Dinh N.-T.
336  57195998372  Dinh N.-T.

In [ ]:
Author_table[Author_table['Author(s) ID']=='56812550600']

Author(s) ID     Author
36   56812550600     Kim Y.
46   56812550600     Kim Y.
91   56812550600     Kim Y.
302  56812550600  Kim Y.-H.
337  56812550600     Kim Y.

So, here we have two authors,  
Dinh N.-T.  
Kim Y.


### h) Highest cited author from India and the university.  

In [ ]:
ans_h = df_final_1[ df_final_1['Country']=='India' ]

ans_h
# these all are indian authors

Author(s) ID                                              Title  Year  \
15    57312879900  Applications of Wireless Sensor Networks and I...  2022   
59    57224473440  Collision‐aware routing using multi‐objective ...  2021   
60    42061434800  Collision‐aware routing using multi‐objective ...  2021   
62    55342182200  Collision‐aware routing using multi‐objective ...  2021   
85    57194224788  Dynamic bargain game theory in the internet of...  2021   
88    36195020800  Dynamic bargain game theory in the internet of...  2021   
89    55342182200  Dynamic bargain game theory in the internet of...  2021   
92    57195678060  Residual energy estimation‐based mac protocol ...  2021   
240   57222245396  Internet of things: Evolution, concerns and se...  2021   
241   57210010055  Internet of things: Evolution, concerns and se...  2021   
242   57193603927  Internet of things: Evolution, concerns and se...  2021   
244   55568517829  Internet of things: Evolution, concerns and se...  2021   
245   57314715300  Internet of things: Evolution, concerns and se...  2021   
246   55757784831  Improving the accuracy of decawave’s uwb mdek1...  2021   
542   56606891300             Smart-farming using internet of things  2020   
543   57197720982             Smart-farming using internet of things  2020   
581   57210615882  Cryptanalysis and improvement of a privacy-pre...  2019   
582   23474214900  Cryptanalysis and improvement of a privacy-pre...  2019   
584   35072405000  Cryptanalysis and improvement of a privacy-pre...  2019   
585   57210819614  A radio channel model for D2D communications b...  2019   
622   55548066000  Sensors driven ai-based agriculture recommenda...  2019   
625   25926970400  Sensors driven ai-based agriculture recommenda...  2019   
710   57205217776  An affinity propagation-based self-adaptive cl...  2019   
1034  57204826670  Intrusion detection in agriculture field based...  2018   
1035  57204827461  Intrusion detection in agriculture field based...  2018   
1036  57204834316  Intrusion detection in agriculture field based...  2018   
1037  57204834318  Intrusion detection in agriculture field based...  2018   
1282  56024601000  A novel scheme for an energy efficient interne...  2015   
1283  50862123000  A novel scheme for an energy efficient interne...  2015   
1284  25628189800  A novel scheme for an energy efficient interne...  2015   

                                           Source title  Cited by  \
15                                              Sensors       NaN   
59                                              Sensors       NaN   
60                                              Sensors       NaN   
62                                              Sensors       NaN   
85                                              Sensors       NaN   
88                                              Sensors       NaN   
89                                              Sensors       NaN   
92                                              Sensors       NaN   
240                                             Sensors      21.0   
241                                             Sensors      21.0   
242                                             Sensors      21.0   
244                                             Sensors      21.0   
245                                             Sensors      21.0   
246                                             Sensors       4.0   
542   Journal of Computational and Theoretical Nanos...       1.0   
543   Journal of Computational and Theoretical Nanos...       1.0   
581                               Sensors (Switzerland)      19.0   
582                               Sensors (Switzerland)      19.0   
584                               Sensors (Switzerland)      19.0   
585                               Sensors (Switzerland)      10.0   
622                               Sensors (Switzerland)      46.0   
625                               Sensors (Switzerland)      46.0   

In [ ]:
ans_h.sort_values(by=['Cited by'], ascending=False).iloc[0:1,:]

Author(s) ID                                              Title  Year  \
1284  25628189800  A novel scheme for an energy efficient interne...  2015   

               Source title  Cited by                 DOI  \
1284  Sensors (Switzerland)     138.0  10.3390/s151128603   

                                                   Link Funding Details  \
1284  https://www.scopus.com/inward/record.uri?eid=2...             NaN   

     Document Type Publication Stage                   Open Access  Source  \
1284       Article             Final  All Open Access, Gold, Green  Scopus   

                     EID                                         University  \
1284  2-s2.0-84947051758  ECE, GNDU Regional Campus, Jalandhar, 144001, ...   

     Country  
1284   India

In [ ]:
key =ans_h.sort_values(by=['Cited by'], ascending=False).iloc[0:1,0:1]
ans_h1 = key['Author(s) ID'].values[0]
ans_h1

'25628189800'

In [ ]:
Author_table[Author_table['Author(s) ID']==ans_h1]

Author(s) ID       Author
1284  25628189800  Malhotra J.

In [ ]:
ans_h2 = df_final_1[ df_final_1['Author(s) ID']==ans_h1 ]
ans_h2

Author(s) ID                                              Title  Year  \
1284  25628189800  A novel scheme for an energy efficient interne...  2015   

               Source title  Cited by                 DOI  \
1284  Sensors (Switzerland)     138.0  10.3390/s151128603   

                                                   Link Funding Details  \
1284  https://www.scopus.com/inward/record.uri?eid=2...             NaN   

     Document Type Publication Stage                   Open Access  Source  \
1284       Article             Final  All Open Access, Gold, Green  Scopus   

                     EID                                         University  \
1284  2-s2.0-84947051758  ECE, GNDU Regional Campus, Jalandhar, 144001, ...   

     Country  
1284   India

In [ ]:
ans_h2['University']

1284    ECE, GNDU Regional Campus, Jalandhar, 144001, ...
Name: University, dtype: object

So, the most cited indian author is  
Malhotra J.  
ECE, GNDU Regional Campus, Jalandhar, Punjab, India.

### i) Comparative year wise article publication analysis of india, china and usa.

In [ ]:
pd.DataFrame(df_final[ df_final_1['Country']=='India'].groupby('Year')['Title'].nunique())
#for india

Title
Year       
2015      1
2018      1
2019      4
2020      1
2021      5
2022      1

In [ ]:
pd.DataFrame(df_final[ df_final_1['Country']=='China'].groupby('Year')['Title'].nunique())
# for China

Title
Year       
2014      3
2015      3
2016      6
2017     11
2018     15
2019     22
2020      8
2021      7
2022      2

In [ ]:
pd.DataFrame(df_final[ df_final_1['Country']=='United States'].groupby('Year')['Title'].nunique())
# for USA

Title
Year       
2014      2
2015      1
2017      7
2018      9
2019      2
2020      4
2021      2
2022      1

### j) Total number of grants given to the field

In [ ]:
df_j = df_final_1.dropna(subset=['Funding Details'])
df_j.groupby(by=['Title']).count().shape[0]

176

So, we find that 176 papers had funding.  


### k) Country wise total number of publication

In [ ]:
d1 = pd.DataFrame(df_final_1.groupby(by=['Country'])['Title'].nunique())
d1

Title
Country                      
Argentina                   1
Australia                  21
Belgium                     6
Bosnia and Herzegovina      1
Brazil                     18
...                       ...
Uganda                      2
United Arab Emirates        3
United Kingdom             12
United States              28
Viet Nam                    1

[63 rows x 1 columns]

Question 2:  
  Co-Author Relationship Prediction and Citation prediction using known machine
learning techniques.  

Ans: Co-Author relationship prediction can be done using the given dataset. One can apply Apriori algorithm on the data set to mine for association rules, which can give good predictions for co-authorship relationship.  

Citation Prediction : here in our data set, see that we don't have enough instances for every author in order train a model to predict number of citations.

In [ ]:
df_final_1.groupby( by=['Author(s) ID']).count().sort_values( by=['Country'], ascending = False)

Title  Year  Source title  Cited by  DOI  Link  Funding Details  \
Author(s) ID                                                                    
24467846500       5     5             5         5    5     5                4   
56812550600       5     5             5         2    5     5                5   
56039568800       5     5             5         5    5     5                4   
8970033500        5     5             5         4    5     5                3   
24467756100       5     5             5         5    5     5                4   
...             ...   ...           ...       ...  ...   ...              ...   
55960576300       1     1             1         1    1     1                1   
55949741600       1     1             1         1    1     1                0   
55946226700       1     1             1         1    1     1                1   
55935399400       1     1             1         1    1     1                0   
9737451300        1     1             1         1    1     1                0   

              Document Type  Publication Stage  Open Access  Source  EID  \
Author(s) ID                                                               
24467846500               5                  5            5       5    5   
56812550600               5                  5            5       5    5   
56039568800               5                  5            5       5    5   
8970033500                5                  5            5       5    5   
24467756100               5                  5            5       5    5   
...                     ...                ...          ...     ...  ...   
55960576300               1                  1            1       1    1   
55949741600               1                  1            1       1    1   
55946226700               1                  1            1       1    1   
55935399400               1                  1            1       1    1   
9737451300                1                  1            1       1    1   

              University  Country  
Author(s) ID                       
24467846500            5        5  
56812550600            5        5  
56039568800            5        5  
8970033500             5        5  
24467756100            5        5  
...                  ...      ...  
55960576300            1        1  
55949741600            1        1  
55946226700            1        1  
55935399400            1        1  
9737451300             1        1  

[1245 rows x 14 columns]

Above are the results.